server/routes/grafana/grafana_routes.py

```
@@ -17,17 +17,20 @@
 def _has_grafana_row(user_id: str) -> Tuple[bool, bool]:
-    """Check if a user_tokens row exists for Grafana (regardless of is_active).
+    """Check if a user_tokens row exists for Grafana visible to user_id (including org-shared).
 
     Returns (row_exists, is_active).
     """
     try:
+        from utils.secrets.secret_ref_utils import _resolve_org, _org_read_predicate
+        org_id = _resolve_org(user_id)
+        predicate, pred_params = _org_read_predicate(user_id, org_id)
         with db_pool.get_admin_connection() as conn:
             with conn.cursor() as cursor:
```

Ruff (0.15.12)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)